### ISSUE TITLE
dist.init_process_group not work

### ISSUE THREAD

**Issue by tensorflowt** [pytorch/pytorch#154102]
### 🐛 Describe the bug

Interface testing：python test.py

```
# test.py
import torch.distributed as dist

dist.init_process_group(
        backend='nccl',
        init_method=f'tcp://10.207.67.15:23456',
        world_size=8,
        rank=0
    )
```
Abnormal return result：
```
[W522 06:29:34.248915854 socket.cpp:204] [c10d] The hostname of the client socket cannot be retrieved. err=-3
[W522 06:29:54.269289676 socket.cpp:204] [c10d] The hostname of the client socket cannot be retrieved. err=-3
```

I checked the hostname information as follows:

![Image](https://github.com/user-attachments/assets/db43cd63-1085-41eb-b471-ea29a9907178)

nccl test results：

![Image](https://github.com/user-attachments/assets/de223b2e-ca18-4c09-848a-ed570e1f89fe)

How can I further locate the problem?

### Versions

Package                           Version       Editable project location
--------------------------------- ------------- ----------------------------
aiohappyeyeballs                  2.5.0
aiohttp                           3.11.13
aiohttp-cors                      0.7.0
aiosignal                         1.3.2
airportsdata                      20250224
annotated-types                   0.7.0
anthropic                         0.49.0
anyio                             4.8.0
argcomplete                       3.6.0
astor                             0.8.1
asttokens                         3.0.0
async-timeout                     5.0.1
attrs                             25.1.0
black                             25.1.0
blake3                            1.0.4
blinker                           1.4
cachetools                        5.5.2
certifi                           2025.1.31
cffi                              1.17.1
charset-normalizer                3.4.1
click                             8.1.8
cloudpickle                       3.1.1
colorful                          0.5.6
compressed-tensors                0.9.1
cryptography                      3.4.8
cuda-bindings                     12.8.0
cuda-python                       12.8.0
datamodel-code-generator          0.28.2
datasets                          3.3.2
dbus-python                       1.2.18
decorator                         5.2.1
decord                            0.6.0
deep_ep                           1.0.0+d5ca449
depyf                             0.18.0
devscripts                        2.22.1ubuntu1
dill                              0.3.8
diskcache                         5.6.3
distlib                           0.3.9
distro                            1.7.0
distro-info                       1.1+ubuntu0.2
einops                            0.8.1
exceptiongroup                    1.2.2
executing                         2.2.0
fastapi                           0.115.11
filelock                          3.17.0
flashinfer-python                 0.2.3
frozenlist                        1.5.0
fsspec                            2024.6.1
genson                            1.3.0
gguf                              0.10.0
google-api-core                   2.24.1
google-auth                       2.38.0
googleapis-common-protos          1.69.1
grpcio                            1.70.0
h11                               0.14.0
hf_transfer                       0.1.9
html5lib                          1.1
httpcore                          1.0.7
httplib2                          0.20.2
httptools                         0.6.4
httpx                             0.28.1
huggingface-hub                   0.31.2
idna                              3.10
importlib_metadata                8.6.1
inflect                           5.6.2
interegular                       0.3.3
ipython                           8.33.0
isort                             6.0.1
jedi                              0.19.2
jeepney                           0.7.1
Jinja2                            3.1.6
jiter                             0.8.2
jsonschema                        4.23.0
jsonschema-specifications         2024.10.1
keyring                           23.5.0
lark                              1.2.2
launchpadlib                      1.10.16
lazr.restfulclient                0.14.4
lazr.uri                          1.0.6
litellm                           1.63.2
llguidance                        0.6.31
lm-format-enforcer                0.10.11
MarkupSafe                        3.0.2
matplotlib-inline                 0.1.7
mistral_common                    1.5.3
modelscope                        1.23.2
mooncake-transfer-engine          0.3.0b6
more-itertools                    8.10.0
mpmath                            1.3.0
msgpack                           1.1.0
msgspec                           0.19.0
multidict                         6.1.0
multiprocess                      0.70.16
mypy-extensions                   1.0.0
nanobind                          2.7.0
nest-asyncio                      1.6.0
networkx                          3.3
ninja                             1.11.1.3
numpy                             1.26.4
nvidia-cublas-cu12                12.4.5.8
nvidia-cuda-cupti-cu12            12.4.127
nvidia-cuda-nvrtc-cu12            12.4.127
nvidia-cuda-runtime-cu12          12.4.127
nvidia-cudnn-cu12                 9.1.0.70
nvidia-cufft-cu12                 11.2.1.3
nvidia-curand-cu12                10.3.5.147
nvidia-cusolver-cu12              11.6.1.9
nvidia-cusparse-cu12              12.3.1.170
nvidia-cusparselt-cu12            0.6.2
nvidia-ml-py                      12.570.86
nvidia-nccl-cu12                  2.21.5
nvidia-nvjitlink-cu12             12.4.127
nvidia-nvtx-cu12                  12.4.127
oauthlib                          3.2.0
openai                            1.65.4
opencensus                        0.11.4
opencensus-context                0.1.3
opencv-python-headless            4.11.0.86
orjson                            3.10.15
outlines                          0.1.11
outlines_core                     0.1.26
packaging                         24.2
pandas                            2.2.3
parso                             0.8.4
partial-json-parser               0.2.1.1.post5
pathspec                          0.12.1
pexpect                           4.9.0
pillow                            11.1.0
pip                               25.0.1
platformdirs                      4.3.6
prometheus_client                 0.21.1
prometheus-fastapi-instrumentator 7.0.2
prompt_toolkit                    3.0.50
propcache                         0.3.0
proto-plus                        1.26.0
protobuf                          5.29.3
psutil                            7.0.0
ptyprocess                        0.7.0
pure_eval                         0.2.3
py-cpuinfo                        9.0.0
py-spy                            0.4.0
pyarrow                           19.0.1
pyasn1                            0.6.1
pyasn1_modules                    0.4.1
pycountry                         24.6.1
pycparser                         2.22
pydantic                          2.10.6
pydantic_core                     2.27.2
Pygments                          2.19.1
PyGObject                         3.42.1
PyJWT                             2.3.0
pynvml                            12.0.0
pyparsing                         2.4.7
python-apt                        2.4.0+ubuntu4
python-dateutil                   2.9.0.post0
python-dotenv                     1.0.1
python-multipart                  0.0.20
pytz                              2025.1
PyYAML                            6.0.2
pyzmq                             26.2.1
ray                               2.43.0
referencing                       0.36.2
regex                             2024.11.6
requests                          2.32.3
rpds-py                           0.23.1
rsa                               4.9
safetensors                       0.5.3
SecretStorage                     3.3.1
sentencepiece                     0.2.0
setproctitle                      1.3.5
setuptools                        75.8.2
sgl-kernel                        0.0.9.post2
sglang                            0.4.5.post3   /sgl-workspace/sglang/python
six                               1.17.0
smart-open                        7.1.0
sniffio                           1.3.1
soundfile                         0.13.1
ssh-import-id                     5.11
stack-data                        0.6.3
starlette                         0.46.0
sympy                             1.13.1
tiktoken                          0.9.0
tokenizers                        0.21.0
tomli                             2.2.1
torch                             2.6.0
torchao                           0.9.0
torchaudio                        2.5.1
torchvision                       0.21.0
tqdm                              4.67.1
traitlets                         5.14.3
transformers                      4.51.1
triton                            3.2.0
typing_extensions                 4.12.2
tzdata                            2025.1
unattended-upgrades               0.1
urllib3                           2.3.0
uvicorn                           0.34.0
uvloop                            0.21.0
virtualenv                        20.29.3
vllm                              0.7.2
wadllib                           1.3.6
watchfiles                        1.0.4
wcwidth                           0.2.13
webencodings                      0.5.1
websockets                        15.0.1
wheel                             0.45.1
wrapt                             1.17.2
xformers                          0.0.28.post3
xgrammar                          0.1.17
xxhash                            3.5.0
yarl                              1.18.3
zipp                              3.21.0

**Comment by JiahuiSun** [comment]
I met the same problem!

![Image](https://github.com/user-attachments/assets/13eaae85-f604-40b3-b456-f234c3569b3a)
